docs/operate/validators/docker-quickstart.mdx

```diff
@@ -0,0 +1,259 @@
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": "*",
+            "Action": [
+                "s3:GetObject",
+                "s3:ListBucket"
+            ],
+            "Resource": [
+                "${BUCKET_ARN}",
+                "${BUCKET_ARN}/*"
+            ]
+        },
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "AWS": "${USER_ARN}"
+            },
+            "Action": [
+                "s3:DeleteObject",
+                "s3:PutObject"
+            ],
+            "Resource": "${BUCKET_ARN}/*"
+        }
+    ]
+}
+```
+
+## Step: 5 Setup docker environment
+
+### Create a directory for each chain you wish to validate
+
+For example, create an ethereum directory and create config.json, docker-compose.yml, and .env
+```bash
+mkdir -p ethereum/hyperlane_db && \
+touch ethereum/config.json ethereum/docker-compose.yml ethereum/.env
+```
+
+### Edit config.json
+
+`customRpcUrls` are a comma separated list of performant RPC endpoints for the chain you wish to support, we recommend using paid providers to avoid rate limiting
+
+`signer.region` and `validator.region` should be adjusted to your AWS region
+
+`signer.id` and `validator.id` should be adjusted to your AWS KMS id you configured in step 3, prefixed with `alias/`
+
+`originchainname` should reflect the chain 
+
+`checkpointsyncer.bucket` should reflect the name of the S3 bucket
+
+`checkpointsyncer.folder` is the name of the folder the validator will use within the bucket (you may use the same bucket for multiple validators, but ensure each folder name is unique per validator)
+
+`reorgperiod` may be different for each chain [**find the reorgperiods**](https://github.com/hyperlane-xyz/hyperlane-registry/tree/main/chains)
+
+```json
+{
+  "chains": {
+    "ethereum": {
+      "customRpcUrls": "https://eth.llamarpc.com,https://rpc.mevblocker.io",
+      "signer": {
+        "region": "us-east-1",
+        "type": "aws",
+        "id": "alias/hyperlane-validator-signer-ethereum"
+      }
+    }
+  },
+  "originchainname": "ethereum",
+  "db": "/mnt/hyperlane_db",
+  "validator": {
+    "id": "alias/hyperlane-validator-signer-ethereum",
+    "type": "aws",
+    "region": "us-east-1"
+  },
+  "checkpointsyncer": {
+    "bucket": "hyperlane-validator-signatures-ethereum",
+    "region": "us-east-1",
+    "type": "s3",
+    "folder": "ethereum"
+  },
+  "reorgperiod": 14,
+  "metrics": "9090"
+}
+```
+
+### Edit .env
+
+```
+AWS_ACCESS_KEY_ID=<Your AWS access key ID>
+AWS_SECRET_ACCESS_KEY=<Your AWS secret access key>
+DB_PATH=./hyperlane_db # ./hyperlane_db
+METRICS_PORT=9091 # Default is 9090, should match port set in config.json
+CONFIG_FILE=./config.json # ./config.json
+SERVICE_NAME=ethereum
+```
+
+### Edit docker-compose.yml
+
+If you wish to run multiple validators, you could reuse the same docker-compose file but create different .env and config.json files per chain. 
```

Comment:
While this docker-compose.yml is a good start, I would like to see it updated to support more than 1 validator.

Ideally, people can do `docker-compose up -d` from the base directory, without having to cd into specific chain dirs`, and that will spin up multiple validators

Reply:
Yes agree, but I guess each operator is going to have different preferences on chains/number of validators etc.

Not sure of the best way to do that. Only thing that came to mind is a separate set of dirs for each chain containing the db folder, config.json and docker-compose.yml, an env in the base directory containing the chains the operator wants to run and Makefile to bring the correct chains up?

Reply:
Yeah each operatore will have a preference and will diverge from this guide in some places anyway, so I think it's sufficient to have a base example showing how to run with 2-3 chains. In the current 1 chain example, it's not so different to the existing docs https://docs.hyperlane.xyz/docs/operate/run-docker-compose

Given the above that describes how to set up the directory for a chain, I think I think you only need to update this docker-compose.yml example

It can work by keeping the existing idea of a dir containing db+env+config for each chain, but the top-level docker-compose specifies a validator for each folder